Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -396,6 +396,89 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             });
         }
 
+        [Test]
+        public void WriteCopy_LocalToSync([Values(true, false)] bool originalEncrypted,
+                                          [Values(true, false)] bool copyEncrypted)
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var originalConfig = new RealmConfiguration(Guid.NewGuid().ToString())
+                {
+                    Schema = new[] { typeof(ObjectIdPrimaryKeyWithValueObject) }
+                };
+                if (originalEncrypted)
+                {
+                    originalConfig.EncryptionKey = TestHelpers.GetEncryptionKey(42);
+                }
+
+                var copyConfig = await GetIntegrationConfigAsync(Guid.NewGuid().ToString());
+                if (copyEncrypted)
+                {
+                    copyConfig.EncryptionKey = TestHelpers.GetEncryptionKey(23);
+                }
+
+                File.Delete(copyConfig.DatabasePath);
+
+                using var originalRealm = GetRealm(originalConfig);
+
+                AddDummyData(originalRealm, true);
+
+                originalRealm.WriteCopy(copyConfig);
+
+                using var copiedRealm = GetRealm(copyConfig);
+
+                Assert.AreEqual(copiedRealm.All<ObjectIdPrimaryKeyWithValueObject>().Count(), DummyDataSize / 2);
```

Comment:
We should use the constrain model for assertions as NUnit recommends in [their official guideline](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html):
```suggestion
                Assert.That(copiedRealm.All<ObjectIdPrimaryKeyWithValueObject>().Count(), Is.EqualTo(DummyDataSize / 2));
```

Reply:
[c8663c3](https://github.com/realm/realm-dotnet/pull/2900/commits/c8663c3eac1e5507a4c5c87bb7b14453b29208d9)